### PR TITLE
[gamekit] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to GameKit.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

Changing any == null or != null to is null and is not null

### CODE REVIEW

File: src/GameKit/GKScore.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public partial class GKScore {
 		// start to differ in future releases (in IOS7 it looks like the older is called, nothing else)
 		public GKScore (string categoryOrIdentifier)
 		{
-			if (categoryOrIdentifier == null)
+			if (categoryOrIdentifier is null)
```
Comment (stephen-hawley) [inline src/GameKit/GKScore.cs:56]: `categoryOrIdentifier` should never be null, right?

File: src/GameKit/GameKit2.cs [hunk 2]
```diff
@@ -201,7 +201,7 @@ public partial class GKVoiceChat {
 		[Obsolete ("Use 'SetMute (bool, string)' method.")]
 		public virtual void SetMute (bool isMuted, GKPlayer player)
 		{
-			if (player == null)
+			if (player is null)
```
Comment (stephen-hawley) [inline src/GameKit/GameKit2.cs:204]: Same - `player` should never be null now.

**Review by stephen-hawley — COMMENTED** [review]
Minor - there are two methods that are checking parameters for null that are no longer able to be null. If they can validly be null, then the argument types should probably be changed to be `type?`.

File: src/GameKit/GKScore.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/GameKit/GKScore.cs:56]: correct, but we have an issue, what if the caller does not have nullability enabled? The compiler won't complain, and the call will be wrong.

**Review by mandel-macaque — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**